### ISSUE TITLE
Does this project support multiple hosts?

### ISSUE THREAD

**Issue by Attacler** [lucaslorentz/caddy-docker-proxy#477]
Hi there,

Does this project support multiple hosts with Docker Swarm?
So if i have 2 nodes running, can i point the domain to the IP of one of the nodes?
Or should it always be one node that is running this project? (so if i have 50 domains, they are all pointing towards the same node which distributes the load to the other nodes)
Next to that, can i run multiple containers with this project inside of it? (ofc im going to use the docker image)

**Comment by francislavoie** [comment]
Yep, that's what the `upstreams` template function does, it expands to the list of available nodes for that container https://github.com/lucaslorentz/caddy-docker-proxy#upstreams

**Comment by Attacler** [comment]
@francislavoie thanks for the fast response.
I`ll rephrase my questions a bit because i think we are talking about differend things here.

If i have 3 servers running together in Docker Swarm, can i use all 3 of the IPs that the servers has? or can i only use the ip that is actually running the caddy-docker-proxy image?

Could i run the caddy-docker-proxy image on all 3 servers at the same time while being part of the same Swarm? (so there is also a bit of loadbalancing happening between 3 containers instead of just 1)

**Comment by Attacler** [comment]
@francislavoie maybe this will clear it up:
![image](https://user-images.githubusercontent.com/42644889/231827452-18722dd9-a269-4b71-ba29-df8d01647b53.png)
I have 3 servers running like you see in the graph.
They are connected to each other trough Docker Swarm (so every 1 is called a node).

If i want to run caddy-docker-proxy on this, would every node need a seperated caddy-docker-proxy running?
Or can i add it once to the swarm network and use that?

Lets say i want to horizontally scale the containers of caddy-docker-proxy, is that possible?

**Comment by Attacler** [comment]
@francislavoie alright thanks for letting me know!
And about the IP part? can i use any IP or should it be only one of the ips?

**Comment by francislavoie** [comment]
You mean for ingress from outside? It depends how you have swarm configured.

**Comment by Attacler** [comment]
@francislavoie thank you very much! that answers my question.
Can i buy you a coffee somewhere?

**Comment by francislavoie** [comment]
I don't, but you could sponsor Matt https://github.com/sponsors/mholt since he works on Caddy full time. I don't think @lucaslorentz has a sponsor link but maybe he should :man_shrugging:
